### PR TITLE
Change forc doc --manifest-path to forc doc --path

### PR DESCRIPTION
**Pull request by alexanderblv**
## Description
Closes https://github.com/FuelLabs/sway/issues/5789

## Checklist

- [ ] I have linked to any relevant issues.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
   - [ ] If my change requires substantial documentation changes, I have [requested support from the DevRel team](https://github.com/FuelLabs/devrel-requests/issues/new/choose)
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [ ] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Review by kayagokalp — CHANGES_REQUESTED** [review]
Hello, thanks for the PR. It seems like you just updated docs and readme, not the command itself.

https://github.com/FuelLabs/sway/blob/24d03f4e9fe13672961bbe696861a99508fa521a/forc-plugins/forc-doc/src/cli.rs#L26

**Comment by alexanderblv** [comment]
> Hello, thanks for the PR. It seems like you just updated docs and readme, not the command itself.
> 
> https://github.com/FuelLabs/sway/blob/24d03f4e9fe13672961bbe696861a99508fa521a/forc-plugins/forc-doc/src/cli.rs#L26

I understand correctly that I have to update `manifest_path` in all sway files?

**Review by IGI-111 — DISMISSED** [review]

**Review by sdankel — CHANGES_REQUESTED** [review]
I would prefer to make `--path` a shorthand for `--manifest-path` (keeping both as valid CLI flags) so as not to break any existing users.

**Comment by alexanderblv** [comment]
> I would prefer to make `--path` a shorthand for `--manifest-path` (keeping both as valid CLI flags) so as not to break any existing users.

So you want to keep both run as valid?
`run: forc doc --manifest-path ./sway-lib-std`
`run: forc doc --path ./sway-lib-std`

I'm not sure if the duplicates will lead to good code performance, but if you think so, then I'll do it.

**Review by sdankel — CHANGES_REQUESTED** [review]
> > I would prefer to make `--path` a shorthand for `--manifest-path` (keeping both as valid CLI flags) so as not to break any existing users.
> 
> So you want to keep both run as valid? `run: forc doc --manifest-path ./sway-lib-std` `run: forc doc --path ./sway-lib-std`
> 
> I'm not sure if the duplicates will lead to good code performance, but if you think so, then I'll do it.

It shouldn't affect the code performance at all. You can use clap's [alias](https://docs.rs/clap/latest/clap/struct.Arg.html#method.alias):

```
    #[clap(long, alias="manifest-path")]
    pub path: Option<String>,
```

Please fix the compilation errors. You need to update `manifest_path` in the other places it is used.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -233,6 +233,7 @@ jobs:
         run: |
           cargo install --locked --debug --path ./forc-plugins/forc-doc
       - name: Build sway-lib-std docs
+        run: forc doc --path ./sway-lib-std
         run: forc doc --manifest-path ./sway-lib-std
```
Comment (kayagokalp) [inline .github/workflows/ci.yml:237]:
```suggestion
        run: |
          forc doc --path ./sway-lib-std
          forc doc --manifest-path ./sway-lib-std
```

File: .github/workflows/gh-pages.yml [hunk 2]
```diff
@@ -50,6 +50,7 @@ jobs:
         run: mdbook build docs/reference
 
       - name: Build Sway std library
+        run: forc doc --path ./sway-lib-std
```
Comment (kayagokalp) [inline .github/workflows/gh-pages.yml:53]:
```suggestion
        run: |
          forc doc --path ./sway-lib-std
          forc doc --manifest-path ./sway-lib-std
```

**Review by kayagokalp — CHANGES_REQUESTED** [review]
Current ci.yml file is invalid, I suggested the correct version of it. But also I think we should not test whole sdk-harness twice just to see that a flag is working with different alias to preserve some valuable CI time. Let's keep the addition to the code to support it for users but remove it from ci. I think @sdankel originally wanted you to add the support to the code not to CI but pinging her in any case.

File: .github/workflows/gh-pages.yml [hunk 3]
```diff
@@ -50,6 +50,7 @@ jobs:
...
         run: forc doc --manifest-path ./sway-lib-std
```
Comment (kayagokalp) [inline .github/workflows/gh-pages.yml:54]:
```suggestion
        run: |
          forc doc --path ./sway-lib-std
          forc doc --manifest-path ./sway-lib-std
```

**Review by kayagokalp — CHANGES_REQUESTED** [review]

File: forc-plugins/forc-doc/src/cli.rs [hunk 4]
```diff
@@ -24,6 +24,7 @@ pub struct Command {
     /// file in the current directory or any parent directory.
     #[clap(long)]
     pub manifest_path: Option<String>,
+    pub path: Option<String>,
```
Comment (kayagokalp) [inline forc-plugins/forc-doc/src/cli.rs:27]: We can probably use, @sdankel's suggestion here:

```suggestion
    #[clap(long, alias="manifest-path")]
    pub path: Option<String>,
```

Comment (kayagokalp) [inline forc-plugins/forc-doc/src/cli.rs:27]: We can probably use, @sdankel's suggestion here:

```suggestion
    #[clap(long, alias="manifest-path")]
    pub path: Option<String>,
```

**Review by kayagokalp — CHANGES_REQUESTED** [review]

**Comment by sdankel** [comment]
I absolutely agree with  @kayagokalp , we should not run forc-doc twice in CI. This code still doesn't compile, so I'm going to close this PR. This isn't really close to what we're looking for and I want to leave the opportunity for another contributor to pick up this task.

**Pull request closed without merging**